npm/private/pnpm.bzl

```diff
@@ -123,10 +123,12 @@ def _convert_pnpm_v5_version_peer_dep(version):
     peer_dep_index = version.find("_")
     if peer_dep_index != -1:
         # if the suffix contains an @version (not just a _patchhash)
-        peer_dep_index = version.find("@", peer_dep_index)
-        if peer_dep_index != -1:
+        peer_dep_at_index = version.find("@", peer_dep_index)
+        if peer_dep_at_index != -1:
             peer_dep = version[peer_dep_index:]
-            version = version[0:peer_dep_index] + utils.sanitize_string(peer_dep)
+            peer_dep = peer_dep.replace("_@", "_at_").replace("@", "_").replace("/", "_").replace("+", "_")
```

Comment:
These now "normalize" the peer dep data so it is the same in all lockfile versions. It does not "sanitize" it.

This can probably be simplified, or maybe only convert old pnpm v5 lockfiles to v6+ and don't "normalize" v6 at all... 🤔